### ISSUE TITLE
Webpack dev server hangs on first build when eslint errors found

### ISSUE THREAD

**Issue by lone-cat** [webpack/eslint-webpack-plugin#293]
### Bug Description

When i recently updated eslint to v.9 with new config and all this bundle (babel, webpack, typesctipt, react, etc..) - webpack dev server with errors from eslint suddenly began to hang on first build. if i run eslint separately it works ok. if i disable eslint plugin - dev server works ok. if errors emission is off - dev server finishes build ok. but if there is some output of eslint - dev server hangs. if i downgrade eslint-webpack-plugin to version 5.0.0 - dev server starts but does not show lint errors on first build, but later rebuild works ok, showing errors normally

### Link to Minimal Reproduction and step to  reproduce

https://github.com/lone-cat/issue-eslint-webpack-plugin

steps:
0. you need docker compose
2. run "init" script to build container and download npm dependencies
3. run "lint" script to see eslint separately works
4. run "run" script, which hangs after showing eslint errors, console responses that i should wait till build process finish if i try visit localhost via browser
eslint-webpack-plugin v.5.0.0. works, but does not emit eslint errors on first build, only on codebase change

### Expected Behavior

build as with previous versions worked for years - emit error and finish build first time, and later on rebuild, nothing special.

### Actual Behavior

webpack serve hangs forever after showing eslint errors, console responses that i should wait till build process finish if i try visit localhost via browser

### Environment

```shell
WSL parameters:
System:
    OS: Linux 6.6 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (12) x64 Intel(R) Xeon(R) w7-3445
    Memory: 89.14 GB / 94.30 GB
  Binaries:
    Node: 24.13.0 - /usr/local/bin/node
    Yarn: 1.22.22 - /usr/local/bin/yarn
    npm: 11.6.2 - /usr/local/bin/npm
  Packages:
    eslint-webpack-plugin: 5.0.2 => 5.0.2 
    html-loader: ^5.1.0 => 5.1.0 
    html-webpack-plugin: ^5.6.3 => 5.6.6 
    ts-loader: ^9.5.2 => 9.5.4 
    webpack: ^5.98.0 => 5.104.1 
    webpack-cli: ^6.0.1 => 6.0.1 
    webpack-dev-server: ^5.2.2 => 5.2.3
```

### Is this a regression?

Yes (please specify version below)

### Last Working Version

v5.0.0 but not fully

**Comment by alexander-akait** [comment]
When you set `failOnError` or `failOnWarning` you can't run compilation, because you literally said - stop and do nothing on error or warning, you should not use this options for dev server/watching

**Comment by lone-cat** [comment]
> Just don't use `failOnError` and `failOnWarning` at all, linter will output problems in console and browser on every change. There are no problems with dev server at all, look at code, we don't use hooks or logic related to eslint plugin - https://github.com/webpack/eslint-webpack-plugin/blob/main/src/index.js#L160

Dear Alexander! Please. Take some time and try out repository i've crated to illustrate the issue. Please. Once again, PLEASE. We're wasting time. In the repository i've gave a link on top of the discussion problem reproduces. And there are NO failOnError or failOnWarning set there at all. You can look throug all webpack.config.js file. I'll just put a link here to line in webpack config where i create eslint plugin object, where you can see which parameters i pass to it: [https://github.com/lone-cat/issue-eslint-webpack-plugin/blob/50ddb4897da1e0227432afb7311aa26df89b5c21/app/webpack.config.cjs#L56](url). 
!! With this config WITHOUT `failOnError` and `failOnWarning` AT ALL webpack devserver HANGS. !!
!! It emits errors but HANGS FOREVER. HANGS. !! 
Please, pay attention, i'm telling that 4th time. PLEASE. Just try to understand i'm writing this for the fourth time. 
As about this project code - i unfortunatly have no time to read every dependency of all my projects for now. I can say that in version prior to 5.0.1 this problem did not exist. And i can find an issue related to error logic of this plugin, which was SUCCESSFULLY closed in v. 5.0.1 but for contributors of this project it is much simplier. But if you need i can also provide link to that issue. Maybe it's related somehow.

**Comment by alexander-akait** [comment]
@lone-cat Sorry for delay, looks like the problem in eslint-webpack-plugin, I will investigate the code and place soon, no problems in dev server, dev server just run server and dev middleware

**Comment by ricardogobbosouza** [comment]
> [@ricardogobbosouza](https://github.com/ricardogobbosouza)
> 
> > The problem is when we use ts-loader.
> 
> Can you clarify?
> 
> > One test I did was change the additionalAssets hook to processAssets
> 
> Yes, `additionalAssets` will be removed in webpack@6, we should use `processAssets`

When using `ts-loader` with `additionalAssets` the error is added but not displayed.

If you remove `ts-loader` the error is displayed

I don't know if the problem is with `additionalAssets` or `ts-loader`

```js
// webpack.config.js
{
  resolve: {
    extensions: [ ".tsx", ".ts", ".js" ]
  },
  mode: 'development',
  plugins: [
    {
      apply(compiler) {
        compiler.hooks.run.tapPromise('test', c => this.run(c));
      },
      async run(compiler) {
        compiler.hooks.compilation.tap('test', async compilation => {
          // 
          compilation.hooks.additionalAssets.tapAsync('test', async function (callback) {
            compilation.errors.push(new Error('Error'));
            callback();
          });
        })
      }
    }
  ],

  module: {
    rules: [
      {
            test: /\.tsx?$/,
            use: {
                loader: 'ts-loader'
            },
            exclude: /node_modules/,
        },
      ]
  },
};
```
